draft/ZEP0009.md

```diff
@@ -0,0 +1,735 @@
+and therefore will follow the active ZEP process. The goal is to encourage more
+exploration by the Zarr community outside of what's currently defined with the
+core specification.
+
+The [third phase](#Phase-3-Future-name-evolution) defines a possible future
+evolution of the naming strategy, largely as a justification for decisions made
+in the first two phases. It is non-normative and can be further discussed in
+future ZEPs.
+
+Revisions to ZEP 0 will be paused until we are clear about the extension
+mechanism and the implications for the role of the ZEP process in the future.
+The ZEP process remains in place for changes to the Zarr v3 core spec including
+the adoption of extensions into the core spec.
+
+## Definitions
+
+- "Core" refers to the Zarr v3 core specification as defined in the document
+  https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html and not
+  necessarily if something is a MUST for implementations.
+- "Metadata" is the Zarr metadata as stored in `zarr.json` files for arrays and
+  groups.
+- "Extensions" as used in this document are components used in the metadata to
+  define and configure how metadata are interpreted by implementations. These
+  components include codecs, data types, chunk key encodings, chunk grids and
+  storage transformers.
+- "Extension points" are locations within metadata where extension-related
+  metadata can be found. Current extension points are listed in the core spec,
+  e.g. `codecs`, `data_type`. As part of this proposal, we intend to add another
+  general-purpose extension point called [`extensions`](#Extension-points).
+- "Extension maintainers" are a person or a team that is responsible for
+  creating and maintaining an extension.## Rollout and next steps
+
+## Phase 1: Immediate clarifications
+
+### Extension naming
+
+🛠️ We propose defining two categories of names for immediate use by extensions:
+raw names and URI-based.
+
+1. **Raw names** MUST be assigned within a central repository and follow the
+   compatibility and versioning [v3 stability
+   policy](https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html#stability-policy).
+   The name assignment is managed through the [`zarr-developers` Github
+   organization](https://github.com/zarr-developers), where each extension is
+   listed and either contains a spec document or links to a spec document.
+   Names are never unassigned or reassigned. The ZSC or by delegation a
+   maintainer team reserves the right to refuse name assignment at its own
+   discretion.
+
+   - **Example:** `zstd`
+   - **Acceptd regex:** `^[a-z0-9-_.]+$`
+
+2. **URI-based names** can be used by anyone without further coordination
+   though the assumption is that users reasonably "own" the URI.
+   URIs have been chosen do to ther self-documenting nature
+   therefore the URL SHOULD resolve to a human-readable explanation of the extension, but
+   implementations SHOULD NOT attempt to resolve the URL during processing.
```

Comment:
I think a URI is self-documenting only if it resolves to some form of documentation, which is not in general true of URIs, so I wouldn't say they have a "self documenting nature", plus I'm not sure about the "self" part -- we want the _zarr extension_ to be documented, not the URI. 

IMO there's 0 value to URIs that don't resolve to anything -- they are just long strings, and everyone will ask why e.g. https is needed if the string doesn't resolve to a resource. But there would be value to a URL that resolves to JSON schema, for example. That's a concrete resource that clients can do something with, and I think STAC uses URLs this way.